ex04.ex

```diff
@@ -39,7 +39,7 @@ defmodule Ex04 do
       [ 1, 2, 3, 4, 5 ]
 
   """
-  def reverse . . . "your code"
+  def reverse([ h | t ]), do: reduce([ h | t ], [], &[ &1 | &2 ])
```

Comment:
## -1

this only works with nonempty lists. 

There's no need to split out [h|t]:

~~~ elixir
  def reverse(list), do: reduce(list, [], &[ &1 | &2 ])
~~~